king_phisher/client/plugins.py

```diff
@@ -543,3 +552,208 @@ class ClientPluginManager(plugins.PluginManagerBase):
 	def __init__(self, path, application):
 		super(ClientPluginManager, self).__init__(path, (application,))
+
+class RepoCache(object):
+	"""
+	RepoCache is used to hold basic information on repos that is to be cached, or pulled from cache.
+	"""
+	__slots__ = ('_id', 'title', 'collections_types', 'url')
+
+	def __init__(self, _id, title, collections_types, url):
+		self._id = _id
+		self.title = title
+		self.collections_types = collections_types
+		self.url = url
+
+	@property
+	def id(self):
+		return self._id
+
+	@property
+	def collections(self):
+		return self.collections_types
+
+	def to_dict(self):
+		repo_cache_values = {
+			'id': self._id,
+			'title': self.title,
+			'collections': self.collections_types,
+			'url': self.url
+		}
+		return repo_cache_values
+
+class CatalogCache(object):
+	"""
+	CatalogCache is used to hold basic information on catalogs that is to be cached or pulled from cache.
+	"""
+	__slots__ = ('_id', 'repos')
+
+	def __init__(self, _id, repos):
+		self._id = _id
+		self.repos = {}
+		for repo in repos:
+			self.repos[repo['id']] = RepoCache(
+				repo['id'],
+				repo['title'],
+				repo['collections'],
+				repo['url']
+			)
+
+	def __getitem__(self, key):
+		return self.repos[key]
+
+	def __iter__(self):
+		for repo in self.repos:
+			yield self.repos[repo]
+
+	@property
+	def id(self):
+		return self._id
+
+	def to_dict(self):
+		catalog_cache_dict = {
+			'id': self._id,
+			'repos': [self.repos[repo].to_dict() for repo in self.repos]
+		}
+		return catalog_cache_dict
+
+class CatalogCacheManager(collections.MutableMapping):
+	"""
+	Manager to handle cache information for catalogs
+	"""
+	def __init__(self, cache_file):
+		self._data = {}
+		self._cache_dict = {}
+		self._cache_cat = {}
+		self._cache_file = cache_file
+
+		if os.path.isfile(cache_file):
+			with open(cache_file) as file_h:
+				try:
+					self._cache_dict = json.load(file_h)
+				except ValueError:
+					self._cache_dict = {}
+
+		if not self._cache_dict or 'catalogs' not in self._cache_dict:
+			self._cache_dict['catalogs'] = {}
+		else:
+			cache_cat = self._cache_dict['catalogs']
+			for catalog_ in cache_cat:
+				self[catalog_] = CatalogCache(
+					cache_cat[catalog_]['id'],
+					cache_cat[catalog_]['repos']
+				)
+
+	def __setitem__(self, key, value):
+		self._data[key] = value
+
+	def __getitem__(self, key):
+		return self._data[key]
+
+	def __delitem__(self, key):
+		del self._data[key]
+
+	def __len__(self):
+		return len(self._data)
+
+	def __iter__(self):
+		for key in self._data.keys():
+			yield key
+
+	def add_catalog_cache(self, cat_id, repos):
+		self[cat_id] = CatalogCache(
+			cat_id,
+			repos
+		)
+
+	def to_dict(self):
+		cache = {}
+		for key in self:
+			cache[key] = self._data[key].to_dict()
+		return cache
+
+	def save(self):
+		self._cache_dict['catalogs'] = self.to_dict()
+		with open(self._cache_file, 'w+') as file_h:
+			json.dump(self._cache_dict, file_h, sort_keys=True, indent=4)
```

Comment:
User the `serializers.JSON` module for dumping and it'll:
1. Handle the pretty printing for you
2. Allow you to automatically serialize Python's datetime objects

Since you're creating all of this data yourself and you are serializing it, I think the `serializers.JSON` class would fit your needs well.